zerver/forms.py

```diff
@@ -154,8 +145,8 @@ def clean_email(self):
             return email
 
         # If no realm is specified, fail
-        realm = get_valid_realm(email)
-        if realm is None:
+        realm = get_realm(resolve_email_to_domain(email))
```

Comment:
This should update the comment above., or split into two conditionals.  Would prefer two conditionals because then coverage will tell us if we're not testing the `invite_required` case.